tools/check-rule-examples.js

```diff
@@ -0,0 +1,150 @@
+ * @returns {Promise<LintMessage[]>} A promise of problems found. The promise will be rejected if an error occurs.
+ */
+async function findProblems(filename) {
+    const text = await readFile(filename, "UTF-8");
+    const problems = [];
+    const ruleExampleOptions = markdownItRuleExample({
+        open(type, code, parserOptions, codeBlockToken) {
+            const languageTag = codeBlockToken.info;
+
+            if (!STANDARD_LANGUAGE_TAGS.has(languageTag)) {
+
+                /*
+                 * Missing language tags are also reported by Markdownlint rule MD040 for all code blocks,
+                 * but the message we output here is more specific.
+                 */
+                const message = `${languageTag
+                    ? `Nonstandard language tag '${languageTag}'`
+                    : "Missing language tag"}: use one of 'javascript', 'js' or 'jsx'`;
+
+                problems.push({
+                    fatal: false,
+                    severity: 2,
+                    message,
+                    line: codeBlockToken.map[0] + 1,
+                    column: codeBlockToken.markup.length + 1
+                });
+            }
+
+            const error = tryParseForPlayground(code, parserOptions);
+
+            if (error) {
+                const message = `Syntax error: ${error.message}`;
+                const line = codeBlockToken.map[0] + 1 + error.lineNumber;
+                const { column } = error;
+
+                problems.push({
+                    fatal: false,
+                    severity: 2,
+                    message,
+                    line,
+                    column
+                });
+            }
+        }
+    });
+
+    // Run `markdown-it` to check rule examples in the current file.
+    markdownIt({ html: true })
+        .use(markdownItContainer, "rule-example", ruleExampleOptions)
+        .render(text);
+    return problems;
+}
+
+/**
+ * Checks the example code blocks in a rule documentation file.
+ * @param {string} filename The file to be checked.
+ * @returns {Promise<LintResult>} The result of checking the file.
+ */
+async function checkFile(filename) {
+    let fatalErrorCount = 0,
+        problems;
+
+    try {
+        problems = await findProblems(filename);
+    } catch (error) {
+        fatalErrorCount = 1;
+        problems = [{
+            fatal: true,
+            severity: 2,
+            message: `Error checking file: ${error.message}`
+        }];
+    }
+    return {
+        filePath: filename,
+        errorCount: problems.length,
+        warningCount: 0,
+        fatalErrorCount,
+        messages: problems
+    };
+}
+
+//------------------------------------------------------------------------------
+// Main
+//------------------------------------------------------------------------------
+
+const patterns = process.argv.slice(2);
+
+(async function() {
+    const globAsync = promisify(glob);
```

Comment:
The latest version of glob exports an async function, so this line (and the `promisify` import at the top) can be removed when glob is updated.